CONTRIBUTING.md

```diff
@@ -29,6 +29,16 @@ If you wish to contribute code, you must sign the [contributor license agreement
 # Open Development
 All contributors follow GitHub’s pull request (PR) workflow to submit changes. Before merging, all pull requests require two approvals.
 
+# Structure of the repository
+
+Our repository is organized as a monorepo, meaning it houses multiple [projects](https://github.com/vmware/versatile-data-kit/tree/main/projects) and [plugins](https://github.com/vmware/versatile-data-kit/tree/main/projects/vdk-plugins) within the same repository.
+
+You can find each [individual project](https://github.com/vmware/versatile-data-kit/tree/main/projects) and [plugin](https://github.com/vmware/versatile-data-kit/tree/main/projects/vdk-plugins) located in their respective directories.
+The design of our monorepo allows for each project and plugin to be developed independently.
+
+What this means for your workflow is that when you want to work on a specific project or plugin, you only need to load that particular one into your IDE, rather than the entire monorepo.
```

Comment:
I'm generally OK with merging the change "as is" as it is still an iterative improvement.

Reply:
>    it's worth mentioning that plugins for examples depend on each other
That's very rare if at all.

 Of course, all plugins depend on vdk-core and vdk-test-utils for testing. But aside from that , I couldn't find another example. 

Even that dependecy is a normal dependency to a 3th party library (e.g they also depend on pandas, click, etc) and should be treated as such. 

> Also that making changes to certain plugins (vdk-core) could lead to issues with other - dependent - plugins
I was wondering if I should mention that, but decided against that since this is 
A) part of CICD which is later stage 
B) it's really part of pre-release phase so it's post merge verification
C) It's specific only for vdk-core build. 

We could mention in vdk-core documentation about that .